### PR TITLE
Setting enhancements

### PR DESCRIPTION
**Pull request by euantorano**
Simplifying the settings package to have a single store and use repositories to load settings.

Also adding a `setting_type` column to the settings table, along with a `default_value` column.

Still need to add some more tests, but at least there are some now...

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,18 +9,29 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
+		"php": ">=7.0.0",
```
Comment (JN-Jones) [inline composer.json]: There seems to be some mixed tabs/spaces here and below

**Comment by euantorano** [comment]
Yeah, not sure what's going on there. I'll fix it tonight.

**Comment by xaoseric** [comment]
I have an idea of how we can make the settings package better, I'll submit a pull later today

**Comment by euantorano** [comment]
Nice one, thanks. Would you mind basing it off this branch, unless it's a completely different approach?

**Comment by Matslom** [comment]
Works fine. We can merge or you plan something more to do with this PR?

**Comment by euantorano** [comment]
Merged, I can always change things if I need to 😄 And this way at least the package has a good working state.

**Pull request merged**